src/main/java/com/app/eece4792mealmaster/services/MealService.java

```
@@ -0,0 +1,86 @@
+package com.app.eece4792mealmaster.services;
+
+import com.app.eece4792mealmaster.dto.MealDto;
+import com.app.eece4792mealmaster.dto.MealFoodDto;
+import com.app.eece4792mealmaster.dto.MealRecipeDto;
+import com.app.eece4792mealmaster.models.*;
+import com.app.eece4792mealmaster.repositories.FoodStockRepository;
+import com.app.eece4792mealmaster.repositories.RecipeRepository;
+import com.app.eece4792mealmaster.repositories.StockItemRepository;
+import com.app.eece4792mealmaster.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Transactional
+@Service
+public class MealService {
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    RecipeRepository recipeRepository;
+
+    @Autowired
+    FoodStockRepository foodStockRepository;
+
+    @Autowired
+    StockItemRepository stockItemRepository;
+
+    public void consume(Long userId, MealDto mealData) {
+        Optional<User> oUser = userRepository.findById(userId);
+        if (!oUser.isPresent()) return; // this should never happen
```

should throw an error